spyrmsd/rmsd.py

```diff
@@ -373,3 +377,191 @@ def rmsdwrapper(
+    symmetry: bool, optional
+        Symmetry-corrected RMSD (using graph isomorphism)
+    center: bool, optional
+        Center molecules at origin
+    minimize: bool, optional
+        Minimised RMSD (using the quaternion polynomial method)
+    strip: bool, optional
+        Strip hydrogen atoms
+    timeout: float, optional
+        After how many seconds to stop the RMSD calculations
+
+    Returns
+    -------
+    List[float]
+        RMSDs
+
+    Notes
+    -----
+    Timeout implemenation inspired by https://superfastpython.com/task-with-timeout-child-process/
+    """
+
+    if not isinstance(mols, list):
+        mols = [mols]
+
+    # RMSD is computed by the child process
+    # The results need to be shared with the parent process
+    # https://docs.python.org/3/library/multiprocessing.html#sharing-state-between-processes
+    result = Value(float)
+    process = Process(
+        target=_rmsd_queue,
+        args=(molref, mols, result, symmetry, center, minimize, strip, cache),
+    )
+
+    process.start()
+    process.join(timeout=timeout)
+
+    # Check if the process finished running successfully
+    if process.exitcode != 0:
+        # Actually terminate the process if it still running
+        if process.is_alive():
+            process.terminate()
+
+        return [np.nan] * len(mols)
+    else:
+        # Retrieve the result from the finished job
+        return result.value
+
+
+def rmsd_parallel(
+    # Should we change this to reflect to possibility of also allowing just 1 molecule as input?
+    molrefs: Union[molecule.Molecule, List[molecule.Molecule]],
+    mols: Union[molecule.Molecule, List[molecule.Molecule]],
+    num_workers: int = 1,
+    symmetry: bool = True,
+    center: bool = False,
+    minimize: bool = False,
+    strip: bool = True,
+    cache: bool = True,
+    timeout: Optional[float] = None,
+) -> Any:
+    """
+    Compute RMSD between two molecules with a timeout.
+
+    Parameters
+    ----------
+    molrefs: Union[molecule.Molecule, List[molecule.Molecule]]
+        Reference molecule
+    mols: Union[molecule.Molecule, List[molecule.Molecule]]
+        Molecules to compare to reference molecule
+    num_workers: int
+        Amount of processor to use for the parallel calculations
+    symmetry: bool, optional
+        Symmetry-corrected RMSD (using graph isomorphism)
+    center: bool, optional
+        Center molecules at origin
+    minimize: bool, optional
+        Minimised RMSD (using the quaternion polynomial method)
+    strip: bool, optional
+        Strip hydrogen atoms
+    timeout: float, optional
+        After how many seconds to stop the RMSD calculations
+
+    Returns
+    -------
+    List[float]
+        RMSDs
+    """
+
+    # Ensure the num_workers is less or equal than the max number of CPUs
+    # Makes MyPy unhappy because os.cpu_count() can return None in some cases (and num_workers is defined as an int)
+    num_workers = min(num_workers, os.cpu_count())
```

Comment:
```suggestion
    num_workers = min(num_workers, os.cpu_count()) if os.cpu_count() is not None else 1
```
Not as clean as I would like, but maybe this will make mypy happy?

Reply:
Earlier I finished making the transition to the shared Value instead of the queue. It did give some more MyPy errors, but i'm almost there. Also, it seems I wasn't the only one having problems with MyPy and multiprocessing.Value, it seems an active issue : https://github.com/python/typeshed/issues/8799#issuecomment-1992699496.

I will push later, hopefully with MyPy happy, otherwise I'll have another go tomorrow!

Reply:
Thanks! Please don't waste too much time on `mypy` issues. It's supposed to be helpful, not annoying. If it becomes too much, we can simply turn it off for the offending line(s) and deal with it another time.

Reply:
Okay, these are the 2 remaining mypy errors in `rmsd.py`:
```
spyrmsd/rmsd.py:493: error: "SynchronizedBase[Any]" has no attribute "value"  [attr-defined]
spyrmsd/rmsd.py:536: error: Value of type variable "SupportsRichComparisonT" of "min" cannot be "int | None"  [type-var]
```
The first one is because of a potential bug that the is also from the discussion I linked in the above comment. 
The second one is the comparison between an `int `and potentially `None`. Even though we patched it, `MyPy `still seems to complain..

Reply:
> Thanks! Please don't waste too much time on `mypy` issues. It's supposed to be helpful, not annoying. If it becomes too much, we can simply turn it off for the offending line(s) and deal with it another time.

I just suppressed the 2 remaining MyPY errors. One of them should hopefully get fixed by MyPy (see the earlier issue I referred to), the other is the problem that os.cpu_count() can return `None` sometimes, which would give problems when using the `min(num_workers, os.cpu_count())`. I tried your patch, but it was still complaining.

Functionally I think it's basically how we want to have, but perhaps some parts of the implementation can still be cleaned up? Besides that, I suppose some tests should be added as well? The timeout feature should be relatively easy to test (by using one of the "problematic" compounds that made me start looking into this in the first place. But i'm not sure how to properly test the multicore feature?

By the way, there is no rush in getting this finished quickly, so please only review it if you actually have the time!

Reply:
Resolving the conversation, since you suppressed the `mypy` errors.